lib/swaps/Swaps.ts

```diff
@@ -104,6 +106,60 @@ class Swaps extends EventEmitter {
     this.pool.on('packet.swapError', this.handleSwapError);
   }
 
+  /**
+   * verify that the order still exists on orderBook
+   */
+  private isOrderOnBook = (pairId: string, orderId: string): boolean => {
+    this.logger.debug('checking order book for ' + pairId + ':' + orderId);
+    return true;
```

Comment:
A followup PR should replace this dummy method with an actual orderbook check. Please add a `TODO` comment

Reply:
Will add a TODO.
Will keep the method as it marks where handling should be and allows Error handling.

Now that we are in good shape with regards to Swap, @sangaman can implement the order book integration.

as a side note: No need to be fanatic about a circular reference.  You need to avoid it with fast moving objects to allow the GC to do a good job. IMHO This is not the case here. Clearly, it is better to have a circular reference here than moving the code to the order book....

Anyway, I will leave it to you to design how the order book changes should be done.  You may like to break the code of Swaps.ts between the different modules.